### PR TITLE
Oracle Web Logic Server Data Connector

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add Oracle Web Logic Server Data Connector

### CODE REVIEW

File: DataConnectors/OracleWebLogicServer/Connector_OracleWebLogicServer_agent.json [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    "connectivityCriterias": [
+        {
+            "type": "IsConnectedQuery",
+            "value": [
+                "OracleWebLogicServerEvent\n            | summarize LastLogReceived = max(TimeGenerated)\n            | project IsConnected = LastLogReceived > ago(30d)"
+            ]
+        }
+    ],
+    "availability": {
+        "status": 1,
```
Comment (v-admahe) [inline DataConnectors/OracleWebLogicServer/Connector_OracleWebLogicServer_agent.json:35]: Status should be 2. Please update the status.

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-admahe** [comment]
@vu-socprime sample data file name should be "OracleWebLogicServer_CL.json". Could you please update the file name.

Thanks,
Aditi Maheshwari

File: Parsers/OracleWebLogicServer/OracleWebLogicServerEvent.txt [hunk 2]
```diff
@@ -0,0 +1,75 @@
+// This function parses Oracle WebLogic Server server.log and access.log
+// Usage Instruction : 
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name and alias as TomcatEvent.
```
Comment (v-admahe) [inline Parsers/OracleWebLogicServer/OracleWebLogicServerEvent.txt:3]: Please update the function name and alias name to "OracleWebLogicServerEvent".

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: DataConnectors/OracleWebLogicServer/Connector_OracleWebLogicServer_agent.json (hunk 1 above)
Comment (vu-socprime) [reply DataConnectors/OracleWebLogicServer/Connector_OracleWebLogicServer_agent.json:35]: updated

File: Parsers/OracleWebLogicServer/OracleWebLogicServerEvent.txt (hunk 2 above)
Comment (vu-socprime) [reply Parsers/OracleWebLogicServer/OracleWebLogicServerEvent.txt:3]: updated

**Comment by vu-socprime** [comment]
> @vu-socprime sample data file name should be "OracleWebLogicServer_CL.json". Could you please update the file name.
> 
> Thanks,
> Aditi Maheshwari

@v-admahe file name was updated

**Review by v-admahe — CHANGES_REQUESTED** [review]

**Comment by v-rucdu** [comment]
@vu-socprime : [Ubiquiti connector json](https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/Ubiquiti/Data%20Connectors/Connector_Ubiquiti_agent.json) is referring to a link (https://aka.ms/sentinel-UbiquitiUnifi-conf) resolving into ubiquiti.conf file, can you please provide it?

I am requesting here as the [Ubiquiti PR](https://github.com/Azure/Azure-Sentinel/pull/1918), is already merged. 

Thanks!!!

File: DataConnectors/OracleWebLogicServer/Connector_OracleWebLogicServer_agent.json [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        }
+    ],
+    "availability": {
+        "status": 2,
+        "isPreview": true
+    },
+    "permissions": {
+        "resourceProvider": [
+            {
+                "provider": "Microsoft.OperationalInsights/workspaces",
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply DataConnectors/OracleWebLogicServer/Connector_OracleWebLogicServer_agent.json:41]: updated

**Comment by vu-socprime** [comment]
> @vu-socprime : [Ubiquiti connector json](https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/Ubiquiti/Data%20Connectors/Connector_Ubiquiti_agent.json) is referring to a link (https://aka.ms/sentinel-UbiquitiUnifi-conf) resolving into ubiquiti.conf file, can you please provide it?
> 
> I am requesting here as the [Ubiquiti PR](https://github.com/Azure/Azure-Sentinel/pull/1918), is already merged.
> 
> Thanks!!!

@v-rucdu - Should we create new PR for this file or add it to current PR? There is content of ubiquiti.conf:
```
<source>
  type udp
  format none
  port 22022
  bind 0.0.0.0
  delimiter "\n"
  tag oms.api.Ubiquiti
</source>

<match oms.api.Ubiquiti>
  type out_oms_api
  log_level info
  num_threads 5
  omsadmin_conf_path /etc/opt/microsoft/omsagent/workspace_id/conf/omsadmin.conf
  cert_path /etc/opt/microsoft/omsagent/workspace_id/certs/oms.crt
  key_path /etc/opt/microsoft/omsagent/workspace_id/certs/oms.key
  buffer_chunk_limit 10m
  buffer_type file
  buffer_path /var/opt/microsoft/omsagent/workspace_id/state/out_oms_api_ubiquiti*.buffer
  buffer_queue_limit 10
  buffer_queue_full_action drop_oldest_chunk
  flush_interval 30s
  retry_limit 10
  retry_wait 30s
  max_retry_wait 9m
</match>
```

**Comment by v-jayakal** [comment]
> > @vu-socprime : [Ubiquiti connector json](https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Solutions/Ubiquiti/Data%20Connectors/Connector_Ubiquiti_agent.json) is referring to a link (https://aka.ms/sentinel-UbiquitiUnifi-conf) resolving into ubiquiti.conf file, can you please provide it?
> > I am requesting here as the [Ubiquiti PR](https://github.com/Azure/Azure-Sentinel/pull/1918), is already merged.
> > Thanks!!!
> 
> @v-rucdu - Should we create new PR for this file or add it to current PR? There is content of ubiquiti.conf:
> 
> ```
> <source>
>   type udp
>   format none
>   port 22022
>   bind 0.0.0.0
>   delimiter "\n"
>   tag oms.api.Ubiquiti
> </source>
> 
> <match oms.api.Ubiquiti>
>   type out_oms_api
>   log_level info
>   num_threads 5
>   omsadmin_conf_path /etc/opt/microsoft/omsagent/workspace_id/conf/omsadmin.conf
>   cert_path /etc/opt/microsoft/omsagent/workspace_id/certs/oms.crt
>   key_path /etc/opt/microsoft/omsagent/workspace_id/certs/oms.key
>   buffer_chunk_limit 10m
>   buffer_type file
>   buffer_path /var/opt/microsoft/omsagent/workspace_id/state/out_oms_api_ubiquiti*.buffer
>   buffer_queue_limit 10
>   buffer_queue_full_action drop_oldest_chunk
>   flush_interval 30s
>   retry_limit 10
>   retry_wait 30s
>   max_retry_wait 9m
> </match>
> ```

@socprime, we will add this file to the corresponding place and will update the link, Thanks.

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**